src/controllers/colormapperjsproxy.cpp

```diff
@@ -0,0 +1,46 @@
+#include <QScriptValueIterator>
+
+#include "controllers/colormapperjsproxy.h"
+
+ColorMapperJSProxy::ColorMapperJSProxy(QScriptEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors)
+    : m_pScriptEngine(pScriptEngine) {
+        m_colorMapper = new ColorMapper(availableColors);
+}
+
+QScriptValue ColorMapperJSProxy::getNearestColor(uint colorCode) {
+    QPair<QRgb, QVariant> result = m_colorMapper->getNearestColor(static_cast<QRgb>(colorCode));
+    QScriptValue jsColor = m_pScriptEngine->newObject();
+    jsColor.setProperty("red", qRed(result.first));
+    jsColor.setProperty("green", qGreen(result.first));
+    jsColor.setProperty("blue", qBlue(result.first));
+    return jsColor;
+}
+
+QScriptValue ColorMapperJSProxy::getNearestValue(uint colorCode) {
+    QPair<QRgb, QVariant> result = m_colorMapper->getNearestColor(static_cast<QRgb>(colorCode));
+    return m_pScriptEngine->toScriptValue(result.second);
+}
+
+QScriptValue ColorMapperJSProxyConstructor(QScriptContext* pScriptContext, QScriptEngine* pScriptEngine) {
+    QMap<QRgb, QVariant> availableColors;
+    DEBUG_ASSERT(pScriptContext->argumentCount() == 1);
+    QScriptValueIterator it(pScriptContext->argument(0));
+    while (it.hasNext()) {
+        it.next();
+        DEBUG_ASSERT(!it.value().isObject());
```

Comment:
is that realistically ever going to happen? Why might it?

Reply:
Could happen if a user passes a non-object in their controller script, e.g. and integer or a string.

Reply:
e.g.:
```js
var mapper = ColorMapper("hello");
``

Reply:
Replaced with a script error.